### PR TITLE
[Bug] The operator on single Sql has affected the session varaiables

### LINKED ISSUE #5570: [Bug] SET_VAR in a single sql affects the Session value
**Describe the bug**
/*+ SET_VAR(..) */ has changed the session value. 
All sql are affected in the same session.

image

**To Reproduce**
1、sql: select/*+ query_timeout = 0*/ 1 + 2;
2、sql: select 1 + 2;
3、sql: select/*+ query_timeout = 10*/ 1 + 2;
4、sql: select 1 + 2;

**Expected behavior**
the second sql and the fourth sql should be the same result;
But the second sql is timeout,and the fourth sql is correct.

### PR DESCRIPTION
**Pull request by Userwhite**
## Proposed changes

1. record the modified session value's initial value.
2. when the sql's execute has finished,set session value to initial value.
3. if it is the set Stmt, won't do step 1、2.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #5570 ) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

### CODE REVIEW

**Comment by Userwhite** [comment]
> ## Proposed changes
> 1. record the modified session value's initial value.
> 2. when the sql's execute has finished,set session value to initial value.
> 
> ## Types of changes
> What types of changes does your code introduce to Doris?
> _Put an `x` in the boxes that apply_
> 
> * [x]  Bugfix (non-breaking change which fixes an issue)
> * [ ]  New feature (non-breaking change which adds functionality)
> * [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
> * [ ]  Documentation Update (if none of the other choices apply)
> * [ ]  Code refactor (Modify the code structure, format the code, etc...)
> 
> ## Checklist
> _Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._
> 
> * [x]  I have created an issue on (Fix #5570 ) and described the bug/feature there in detail
> * [ ]  Compiling and unit tests pass locally with my changes
> * [x]  I have added tests that prove my fix is effective or that my feature works
> * [ ]  If these changes need document changes, I have updated the document
> * [ ]  Any dependent changes have been merged

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java [hunk 1]
```diff
@@ -253,7 +253,10 @@ public void execute() throws Exception {
     // Exception:
     //  IOException: talk with client failed.
     public void execute(TUniqueId queryId) throws Exception {
-
+        // origin value init
+        VariableMgr.setIsSingleSetVar(false);
+        VariableMgr.clearMapSessionOriginValue();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:258]: VariableMgr is a thread safe class.
But `execute()` can be called by multi threads

Comment (morningman) [reply fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:258]: So  `isSingleSetVar` and `sessionOriginValue` should be set in `SessionVariables` instance.

Comment (Userwhite) [reply fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java]: Got it and Fixed. sorry for my fault.

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 2]
```diff
@@ -124,6 +124,12 @@
     public static final long DEFAULT_INSERT_VISIBLE_TIMEOUT_MS = 10_000;
     public static final long MIN_INSERT_VISIBLE_TIMEOUT_MS = 1000; // If user set a very small value, use this value instead.
 
+    // session origin value
+    public static final Map<Field, String> sessionOriginValue = new HashMap<Field, String>();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:128]: Why using `static`?

Comment (Userwhite) [reply fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java]: fixed

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**